apollo-router/src/services/http/service.rs

```diff
@@ -0,0 +1,390 @@
+        } else {
+            builder.wrap_connector(http_connector)
+        };
+
+        let http_client = hyper::Client::builder()
+            .pool_idle_timeout(POOL_IDLE_TIMEOUT_DURATION)
+            .http2_only(http2 == Http2Config::Http2Only)
+            .build(connector);
+        Ok(Self {
+            client: ServiceBuilder::new()
+                .layer(DecompressionLayer::new())
+                .service(http_client),
+            service: Arc::new(service.into()),
+        })
+    }
+}
+
+pub(crate) fn generate_tls_client_config(
+    tls_cert_store: Option<RootCertStore>,
+    client_cert_config: Option<&TlsClientAuth>,
+) -> Result<rustls::ClientConfig, BoxError> {
+    let tls_builder = rustls::ClientConfig::builder().with_safe_defaults();
+    Ok(match (tls_cert_store, client_cert_config) {
+        (None, None) => tls_builder.with_native_roots().with_no_client_auth(),
+        (Some(store), None) => tls_builder
+            .with_root_certificates(store)
+            .with_no_client_auth(),
+        (None, Some(client_auth_config)) => tls_builder.with_native_roots().with_client_auth_cert(
+            client_auth_config.certificate_chain.clone(),
+            client_auth_config.key.clone(),
+        )?,
+        (Some(store), Some(client_auth_config)) => tls_builder
+            .with_root_certificates(store)
+            .with_client_auth_cert(
+                client_auth_config.certificate_chain.clone(),
+                client_auth_config.key.clone(),
+            )?,
+    })
+}
+
+impl tower::Service<HttpRequest> for HttpClientService {
+    type Response = HttpResponse;
+    type Error = BoxError;
+    type Future = BoxFuture<'static, Result<Self::Response, Self::Error>>;
+
+    fn poll_ready(&mut self, cx: &mut std::task::Context<'_>) -> Poll<Result<(), Self::Error>> {
+        self.client
+            .poll_ready(cx)
+            .map(|res| res.map_err(|e| Box::new(e) as BoxError))
+    }
+
+    fn call(&mut self, request: HttpRequest) -> Self::Future {
+        let HttpRequest {
+            mut http_request,
+            context,
+        } = request;
+
+        let schema_uri = http_request.uri();
+        let host = schema_uri.host().unwrap_or_default();
+        let port = schema_uri.port_u16().unwrap_or_else(|| {
+            let scheme = schema_uri.scheme_str();
+            if scheme == Some("https") {
+                443
+            } else if scheme == Some("http") {
+                80
+            } else {
+                0
+            }
+        });
+
+        let path = schema_uri.path();
+
+        let http_req_span = tracing::info_span!("http_request",
+            "otel.kind" = "CLIENT",
+            "net.peer.name" = %host,
+            "net.peer.port" = %port,
+            "http.route" = %path,
+            "http.url" = %schema_uri,
+            "net.transport" = "ip_tcp",
+            //"apollo.subgraph.name" = %service_name,
+            //"graphql.operation.name" = %operation_name,
```

Comment:
Should this be deleted or should we be using the `service_name` from line 236? Does it matter that the operation name is lost from the span?

Reply:
I think we should remove it, because the goal is to reuse that service in other parts of the router, like uplink